src/openrct2/cmdline/ParkInfoCommands.cpp

```diff
@@ -0,0 +1,145 @@
+ * Copyright (c) 2014-2021 OpenRCT2 developers
+ *
+ * For a complete list of all authors, please refer to contributors.md
+ * Interested in contributing? Visit https://github.com/OpenRCT2/OpenRCT2
+ *
+ * OpenRCT2 is licensed under the GNU General Public License version 3.
+ *****************************************************************************/
+
+#include "../Context.h"
+#include "../FileClassifier.h"
+#include "../ParkImporter.h"
+#include "../core/Console.hpp"
+#include "../core/FileStream.h"
+#include "../core/Path.hpp"
+#include "../object/ObjectRepository.h"
+#include "CommandLine.hpp"
+
+// clang-format off
+static constexpr const CommandLineOptionDefinition NoOptions[]
+{
+    OptionTableEnd
+};
+
+static exitcode_t HandleObjectsInfo(CommandLineArgEnumerator *argEnumerator);
+
+const CommandLineCommand CommandLine::ParkInfoCommands[]{
+    // Main commands
+    DefineCommand("objects", "<savefile>", NoOptions, HandleObjectsInfo),
+
+    CommandTableEnd
+};
+// clang-format on
+
+static exitcode_t HandleObjectsInfo(CommandLineArgEnumerator* argEnumerator)
+{
+    exitcode_t result = CommandLine::HandleCommandDefault();
+    if (result != EXITCODE_CONTINUE)
+    {
+        return result;
+    }
+
+    // Get the source path
+    const utf8* rawSourcePath;
+    if (!argEnumerator->TryPopString(&rawSourcePath))
+    {
+        Console::Error::WriteLine("Expected a source save file path.");
+        return EXITCODE_FAIL;
+    }
+
+    auto sourcePath = Path::GetAbsolute(rawSourcePath);
+
+    auto context = OpenRCT2::CreateContext();
+    context->Initialise();
+
+    auto stream = OpenRCT2::FileStream(sourcePath, OpenRCT2::FILE_MODE_OPEN);
+    ClassifiedFileInfo info;
+    if (!TryClassifyFile(&stream, &info))
+    {
+        Console::Error::WriteLine("Unable to detect file type");
+        return EXITCODE_FAIL;
+    }
+
+    if (info.Type != FILE_TYPE::PARK && info.Type != FILE_TYPE::SAVED_GAME && info.Type != FILE_TYPE::SCENARIO)
+    {
+        Console::Error::WriteLine("Invalid file type.");
+        return EXITCODE_FAIL;
+    }
+
+    auto& objectRepository = context->GetObjectRepository();
+    std::unique_ptr<IParkImporter> parkImporter;
+    if (info.Type == FILE_TYPE::PARK)
+    {
+        parkImporter = ParkImporter::CreateParkFile(objectRepository);
+    }
+    else if (info.Version <= FILE_TYPE_S4_CUTOFF)
+    {
+        // Save is an S4 (RCT1 format)
+        parkImporter = ParkImporter::CreateS4();
+    }
+    else
+    {
+        // Save is an S6 (RCT2 format)
+        parkImporter = ParkImporter::CreateS6(objectRepository);
+    }
+    auto loadResult = parkImporter->LoadSavedGame(sourcePath.c_str());
+
+    Console::WriteLine("File contains the following objects: ");
+    Console::WriteLine();
+
+    const std::array<std::string, 17> typeToName = {
+        "Ride",        "SmallScenery", "LargeScenery", "Walls",           "Banners",          "Paths",
```

Comment:
Do we want these in English only?

Reply:
All the other console commands are English only.

Reply:
That isn’t really a good reason. But if this is indeed to be used as input for other programs, then it makes sense to have it in English only.

Reply:
These are power user commands, so anyone using them is likely to need to understand English anyway. I could be wrong, but I doubt the time and effort is worth it (not just dev time, but also translators time).

Reply:
I am a power user, but the difference between reading English and Dutch is quite clear - the latter takes significantly less focus and energy.

Reply:
Are we happy then with the pr overall?